woocommerce/woocommerce-ios
# Product Subscriptions: Check payment sync month to allow one time shipping

Closes: #11178


## Description

#### What? 

When synchronise renewals is turned on (https://woo.com/document/subscriptions/renewal-synchronisation/#section-2) the product JSON response's `_subscription_payment_sync_date` field can have string or a dictionary as value based on the subscription setting.

If subscription interval is yearly:
```json
{
  "id": 21725,
  "key": "_subscription_payment_sync_date",
  "value": {
    "day": 0,
    "month": "01"
  }
}
```

If subscription interval is other than yearly:
```json
{
  "id": 4210,
  "key": "_subscription_payment_sync_date",
  "value": "3"
}
```

#### Why?

Currently, we are ignoring the dictionary format response and enabling the one time shipping setting even when synchronise renewals setting is ON for the product. 

We need to parse both JSON formats and disable the one time shipping setting if synced renewal is turned on. 

#### How?
- Parse the `_subscription_payment_sync_date` in dictionary format and store the `month` value if available.
- Check the payment sync month value before supporting one time shipping setting. 

## Testing instructions

Subscription with synced renewals
- Enable synced renewals option from the plugin's settings (https://woo.com/document/subscriptions/renewal-synchronisation/#section-2)
- Open the product in wp-admin
- Set the subscription interval as yearly.
- Enable synced renewals for the subscription. 
- Open the subscription in the app.
- Open the shipping options.
- Confirm the "one time shipping" option is disabled.

Sample config
<img width="1214" alt="Screenshot 2023-11-29 at 1 47 44 PM" src="https://github.com/woocommerce/woocommerce-ios/assets/524475/4d4957a8-206d-4f8c-9404-00cb1eb3336a">

## Screenshots

<img src="https://github.com/woocommerce/woocommerce-ios/assets/524475/dd174fa3-abe4-4aa1-bacb-e296d01928ae" width="320"/>

---
